lib/ansible/modules/packaging/os/homebrew_cask.py

```diff
@@ -552,7 +581,23 @@ def _install_current_cask(self):
         )
 
         cmd = [opt for opt in opts if opt]
-        rc, out, err = self.module.run_command(cmd)
+
+        rc, out, err = '', '', ''
+
+        if self.sudo_password:
+            sudo_askpass_file = tempfile.NamedTemporaryFile()
+            sudo_askpass_file.write("#!/bin/sh\n\necho '" + self.sudo_password + "'\n")
+            os.chmod(sudo_askpass_file.name, 0o700)
+            sudo_askpass_file.file.close()
+
+            rc, out, err = self.module.run_command(
+                cmd,
+                environ_update={'SUDO_ASKPASS': sudo_askpass_file.name}
+            )
+
+            sudo_askpass_file.close()
```

Comment:
My only concern with this in general is if `run_command()` fails or raises an exception, the file may not get deleted, leaving the `sudo` password in clear text on disk. Use `module.add_cleanup_file()` to add the tempfile to the list of files that will be cleaned up up in the event of failure.